### ISSUE TITLE
False positive opening_braces for multiline if with predicates

### ISSUE THREAD

**Issue by nelsyeung** [realm/SwiftLint#3476]
### The bug

False positive `opening_braces` for multiline if with predicates:
```bash
warning: Opening Brace Spacing Violation: Opening braces should be preceded by a single space and on the same line as the declaration. (opening_brace)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Reproduction file

```swift
func fooFun() {
    let foo: String? = "foo"
    let bar = ["bar"]

    if
        let foobar = foo,
        let barIndex = bar.firstIndex(where: { $0 == foobar })
    { // This line shows the above warning. (This comment doesn't need to be here to trigger it)
        print(barIndex)
    }
}
```

Note that this will not show a warning:
```swift
func fooFun() {
    let foo: String? = "foo"
    let bar: String? = "bar"

    if
        let foooo = foo,
        let barrr = bar
    {
        print(foooo + barrr)
    }
}
```

### Environment

* SwiftLint version: 0.42.0
* Xcode version: Xcode 12.3 Build version 12C33
* Installation method used: Swift Package Manager
* Paste your configuration file:

```yml
identifier_name:
  excluded:
    - id
    - ip
    - to

trailing_comma:
  mandatory_comma: true
```

**Comment by Vitalii-Gozhenko** [comment]
Same problem for me

**Comment by erick2red** [comment]
I get it in this file too, on line 3

```swift
let result = myFuncWithWrappedParameters(one,                                 |
                                         two,                                 |
                                         { (three, four) in three + four })   |
```

_The pipe is showing the margin line_
